### DISCUSSION TITLE
can i set description in pydantic model for openapi?

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#7974] [Questions] [answered]
can i set description in pydantic model for openapi? like a:
```python
class User(BaseModel):
    id: int
    class Config:
        description = 'some desc'
```

**Comment by euri10** [comment]
If  I understand correctly what you want then hte answer is yes, just add a docstring like in the `MainModel` you can see here:
https://pydantic-docs.helpmanual.io/usage/schema/

**Comment by ghost** [comment]
@euri10 i mean this 
![DeepinScreenshot_выберите-область_20191029202620](https://user-images.githubusercontent.com/1178208/67782322-4ce78300-fa71-11e9-8d37-45ae17c07ff1.png)

**Comment by euri10** [comment]
I didn't understand

**Comment by ghost** [comment]
@euri10 
![DeepinScreenshot_выберите-область_20191029203742](https://user-images.githubusercontent.com/1178208/67783381-d64b8500-fa72-11e9-9c95-50a0042410c5.png)

**Comment by euri10** [comment, marked as answer]
ok you need to use response_description in the decorator
should you use an IDE like Pycharm you'd have it on autocomplete ;)

On Tue, Oct 29, 2019 at 4:38 PM foozzi <notifications@github.com> wrote:

> @euri10 <https://github.com/euri10>
> [image: DeepinScreenshot_выберите-область_20191029203742]
> <https://user-images.githubusercontent.com/1178208/67783381-d64b8500-fa72-11e9-9c95-50a0042410c5.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tiangolo/fastapi/issues/667?email_source=notifications&email_token=AAINSPXRLMBGAIXEFRC4MTTQRBKGDA5CNFSM4JGIY5HKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECQ7GVI#issuecomment-547484501>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAINSPUV5HBTRNOR42X2MATQRBKGDANCNFSM4JGIY5HA>
> .
>


-- 
benoit barthelet
http://pgp.mit.edu/pks/lookup?op=get&search=0xF150E01A72F6D2EE

**Comment by ghost** [comment]
@euri10 it's nvim, okay thx)

**Comment by euri10** [comment]
Ha kudos for using vim, you maybe know, maybe not, you can have same key
bindings than vim inside pycharm, that is what I'm using, you get vim
navigation and edition superpowers AND fancy pycharm goodies 🤣

**Comment by ghost** [comment]
@euri10 i know, but no time for that )

hm, last question... how to i build schema from json array like: {"user":{'token':'str','rtoken':'str'}, 'some_key':'some_val'}? 
```python
class Model(BaseModel):
    some_key: str
    user: List[Item] <-- but how to validation this list?
```

p.s. thanks and sorry for offtop!

**Comment by prostomarkeloff** [comment]
> @euri10 i know, but no time for that )
> 
> hm, last question... how to i build schema from json array like: {"user":{'token':'str','rtoken':'str'}, 'some_key':'some_val'}?
> 
> ```python
> class Model(BaseModel):
>     some_key: str
>     user: List[Item] <-- but how to validation this list?
> ```
> 
> p.s. thanks and sorry for offtop!

https://github.com/samuelcolvin/pydantic/blob/master/pydantic/validators.py#L205

**Comment by tiangolo** [comment]
Thanks @euri10 for the help!

@foozzi I think I didn't understand the last question.

You mention an array but the example only has JSON objects. Maybe you can write an example of your JSON, a minimal app example and what is not working properly?

**Comment by ghost** [comment]
@tiangolo 
in endpoint:
```python
def send_sms_code(response_model=UserSchema.UserPayload, db: Session = Depends(get_db)):
    payload = {'user':
        {
            'number': 380661111111,
            'sms_id': 1212,
            'issual_time': 1212,
            'expire_time': 121212
        },
            'jwtToken': 'dfsadf',
            'refreshToken': 'rgsfg3434'
        }
    return payload
```
need to describe returned payload in openapi (response_model)

in schema:
```python
class UserPayload(BaseModel):
    user: List[str, int] --> need validate 'number', 'jwtToken' and each other
```
but i can't validate all items in json (schema)

**Comment by tiangolo** [comment]
I imagine you solved your problem, so thanks for closing the issue @foozzi . :+1: 

For completeness, there's a whole new section in the docs about this: https://fastapi.tiangolo.com/tutorial/schema-extra-example/
